src/app/modules/truenas-connect/components/truenas-connect-spinner/truenas-connect-spinner.component.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  ChangeDetectionStrategy, Component, OnDestroy, AfterViewInit,
+} from '@angular/core';
+
+@Component({
+  selector: 'ix-truenas-connect-spinner',
+  standalone: true,
+  templateUrl: './truenas-connect-spinner.component.html',
+  styleUrl: './truenas-connect-spinner.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TruenasConnectSpinnerComponent implements AfterViewInit, OnDestroy {
```

Comment:
Not sure if we wanna code in animations so specifically. Maybe SVG can handle the animation on it's own and it can be used as a static asset. We could have different SVGs for different states and switch between them based on the component state.